\documentclass{bioinfo}
\copyrightyear{2014}
\pubyear{2014}

\usepackage{fixltx2e}
\usepackage{natbib}
\bibliographystyle{apalike}

\begin{document}
\firstpage{1}

\title[short Title]{DNAism: Exploring genomic datasets on the web with Horizon Charts. }

\author[Sample \textit{et~al}]{David Rio Deiros\,$^{1,}$\footnote{to whom correspondence should be addressed},
Richard A. Gibbs\,$^{1,2}$ and Jeffrey Rogers\, $^{1,2}$}

\address{$^{1}$Human Genome Sequencing Center, Baylor College of Medicine, Houston, TX.\\
$^{2}$Department of Molecular and Human Genetics, Baylor College of Medicine, Houston, TX.}

\history{Received on XXXXXX ; revised on September YYYYYYY; accepted on ZZZZZZZZ}

\editor{Associate Editor: Name Here.}

\maketitle

\begin{abstract}

\section{Summary:}

Computational biologists daily face the need to explore massive amounts of
genomic data.  New visualization techniques help researchers navigate and
understand these big data. Horizon Charts are a
relatively new visualization method that, under the right
circumstances, maximizes data density without losing
graphical perception. This visualization technique has been successfully
applied to understand multi-metric time series
data. We have adapted an existing Javascript
library (Cubism\footnote{https://github.com/square/cubism.}) that implements
Horizon Charts for the time series domain so that it works effectively with
genomic datasets. We call this new library DNAism. 
%For big datasets we provide
%a backend server implemented in Python that relies on tabix to
%efficiently retrieve only the necessary data points for the visualizations.

\section{Availability and implementation:}
Source code, documentation and usage examples can be found at:
http://dnaism.github.com.

\section{Contact:} \href{deiros@bcm.edu}{deiros@bcm.edu}
\end{abstract}

\section{Introduction}

Sharing and communicating about large and intricate datasets produced by
Next-Gen sequencing can be a challenging task. Visual channels are an effective
way to explore data. However, the accelerating increase in data quantity is
pushing the limits of current approaches for representing these datasets
visually without sacrificing accuracy or graphical perception.  Data volume is
increasing vertically as throughput per sample increases, and horizontally as
studies involving large numbers of subjects become more feasible.  Thus, more
effective visualization techniques are needed to understand the most
challenging Next-Gen sequencing datasets.

Horizon Charts~\citep{time-in-the-horizon} have proved to be an 
effective~\citep{2009-horizon}
visualization approach when working with multi-metric time series 
encoded data.
On the other hand, BED\footnote{http://genome.ucsc.edu/FAQ/FAQformat.html}
files are the gold standard for capturing genomic metrics in the Next-Gen
sequencing domain. In time series, metrics are monitored over time, however,
BED files use genomic coordinates. We have adapted a time series Javascript
library to the genomic domain. We call our new library DNAism.


\begin{figure*}
\centerline{\includegraphics[width=1\textwidth,natwidth=485.65,natheight=187.53]{figure.pdf}}
\caption{
Horizon Charts emerge from applying a set of changes to traditional line
charts~\textbf{(A)}. We start by coloring the underlying area of the line chart,
using different hues for positive and negative values. Next, we divide the
graph in bands and apply a gradient of color that increases along with the
quantitative value of the variable we are investigating~\textbf{(B)}. In the
next step, negative values are flipped over the baseline~\textbf{(C)},
effectively reducing the vertical space by two fold. In a final step, bands are
collapsed making all of them start at the baseline and providing another level
of space reduction~\textbf{(D)}.  Visualization of read depth across whole
genome sequences from 24 rhesus macaque samples (30x coverage) for genomic
region Chr17:1.1M-1.2M~\textbf{(E)}. In this snapshot we can appreciate the
strengths of the library and visualization technique: First, differences across
patterns are easily observed.  Second, comparing the metric values across all
samples is simple and effective (There are regions consistently
underrepresented across all the samples and sample 32510 has low coverage
across the whole genomic region). Note that the metric we are exploring in 
this example, read depth, cannot contain negative values. Therefore, only 
green hues appear in~\textbf{(E)}.
%We are
%effectively conveying information from 500k data points (notice there is a
%downsampling process to map input values to the available area for the
%visualization).
}\label{fig:01}
\end{figure*}

\section{Implementation}

Contrary to time series data, in genomic datasets, the variable under study is
associated with chromosomal coordinates instead of timestamps. We have modified
an existing time series data visualization library (based on D3~\citep{2011-d3})
called Cubism to support genome coordinate data. This makes DNAism
a flexible and effective tool to explore multi-sample genomic datasets using
Horizon charts.

To visualize genomic datasets, we have modified most of the components of the
original Cubism library.  The two major ones being 'context' and
'source'.  The 'context' component performs several functions, most
importantly, it defines the region of the genome we want to explore. This
component also specifies, in pixels, how much vertical space we have available
for the visualization.  The 'source' component parses the genomic raw data and
generates the data points necessary for visualization. Our library provides two
sources: 'bedfile' and 'bedserver'. Once the sources are created we can use the
metric component to instantiate metrics pointing to specific samples.  Finally,
the horizon component encapsulates the functionality necessary to create the
visual elements. Figure~\ref{fig:01} illustrates visualization of sample data.

One of the crucial features of DNAism is the ability to efficiently parse and
load the genomic data for visualization. We have provided two alternatives via
the bedfile and the bedserver sources. A bedfile is a simple solution that
loads all the genomic data in memory and returns the relevant data when
queried.  However, this approach is not adequate for larger datasets,
especially those involving multi-sample data. To handle such cases, the
bedserver source can be used. A bedserver is a dedicated server that
implements a RESTful API interface. The client's code running in the browser
can send queries to this server to obtain the data of interest.  The server
uses pre-indexed~\citep{tabix-li} data to speed up random access and returns
only the necessary information for the visualization back to the client.
Hence, this approach becomes much more scalable even with large sized genomic
data sets. We have implemented bedserver as a Python package.

Our source code has a decoupled interface that facilitates the extension of
this library to new data sources. DNAism is data agnostic. As a result, users
can create new sources to capture their specific backend peculiarities.


\section{Results}

We introduce the genomics community to a powerful visualization technique
previously used in the time series data domain. This method facilitates
identification of abnormal patterns across multi-sample datasets.  In addition, this
approach helps to explore and visualize high density datasets more effectively,
thereby, helping the researchers to understand the data easily.

Our library keeps the effective and elegant interface of the original,
while allowing users to leverage its power for genomic data. By providing a
library, we maintain flexibility regarding how to use these resources. Users
can build full applications or use the library within their existing ones.

The companion lightweight server will facilitate the exploration of large
genomic datasets without affecting user experience by using indexed datasets.
Alternatively, users can create their own data sources to reflect the details
of their own environments.

\paragraph{Funding\textcolon} NIH(NCRR) R24OD011173 and NIH grant 2U54HG003273.

\bibliography{document}

\end{document}
